packages/lit-dev-content/src/util/hydration-helpers.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Calls a function on idleCallback or in 200ms if on Safari. _Note:_ 200ms is
+ * what Astro uses for their idle callback fallback.
+ * @param hydrationCallback Function to be called on idleCallback.
+ */
+export const onIdle = (hydrationCallback: () => void) => {
+  if ('requestIdleCallback' in window) {
+    (window as any).requestIdleCallback(hydrationCallback);
+  } else {
+    // Safari doesn't have idle callback so just wait 200ms
+    setTimeout(hydrationCallback, 200);
+  }
+};
+
+/**
+ * Calls a given function when window.matchMedia matches for the first time.
+ *
+ * @param queryString Media query string to trigger the callback.
+ * @param hydrationCallback Callback to be called when the given media query matches.
+ */
+export const onMediaQuery = (
+  queryString: string,
+  hydrationCallback: () => void
+) => {
+  const query = window.matchMedia(queryString);
+  const onQuery = (e: MediaQueryList | MediaQueryListEvent) => {
+    if (e.matches) {
+      hydrationCallback();
+      query.removeEventListener('change', onQuery);
```

Comment:
Instead of removing the event listener, can pass `{once: true}` to the `addEventListener` call.

[Reference](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#:~:text=to%20false.-,once,-A%20boolean%20value)

Reply:
Can't do this because it might call the listener but not match and then remove itself

Reply:
Oops! Missed that behavior.